data-protocols/dsp/dsp-negotiation/dsp-negotiation-transform/src/main/java/org.eclipse.edc.protocol.dsp.negotiation.transform/from/JsonObjectFromContractNegotiationEventMessageTransformer.java

```diff
@@ -46,35 +45,28 @@ public JsonObjectFromContractNegotiationEventMessageTransformer(JsonBuilderFacto
     }
 
     @Override
-    public @Nullable JsonObject transform(@NotNull ContractNegotiationEventMessage object, @NotNull TransformerContext context) {
+    public @Nullable JsonObject transform(@NotNull ContractNegotiationEventMessage eventMessage, @NotNull TransformerContext context) {
         var builder = jsonFactory.createObjectBuilder();
-        builder.add(JsonLdKeywords.ID, String.valueOf(UUID.randomUUID()));
-        builder.add(JsonLdKeywords.TYPE, DSPACE_NEGOTIATION_EVENT_MESSAGE);
-
-        builder.add(DSPACE_NEGOTIATION_PROPERTY_PROCESS_ID, object.getProcessId());
-        builder.add(DSPACE_NEGOTIATION_PROPERTY_CHECKSUM, object.getChecksum());
-
-        var eventType = eventType(object, context);
-        if (eventType == null) {
-            return null;
-        }
+        builder.add(ID, randomUUID().toString());
+        builder.add(TYPE, DSPACE_NEGOTIATION_EVENT_MESSAGE);
 
-        builder.add(DSPACE_NEGOTIATION_PROPERTY_EVENT_TYPE, eventType);
+        builder.add(DSPACE_NEGOTIATION_PROPERTY_PROCESS_ID, eventMessage.getProcessId());
 
+        builder.add(DSPACE_NEGOTIATION_PROPERTY_EVENT_TYPE, eventType(eventMessage));
 
         return builder.build();
     }
 
-    private String eventType(ContractNegotiationEventMessage message, TransformerContext context) {
+    @NotNull
+    private String eventType(ContractNegotiationEventMessage message) {
         switch (message.getType()) {
             case ACCEPTED:
                 return DSPACE_NEGOTIATION_PROPERTY_EVENT_TYPE_ACCEPTED;
             case FINALIZED:
                 return DSPACE_NEGOTIATION_PROPERTY_EVENT_TYPE_FINALIZED;
             default:
-                context.reportProblem(String.format("Could not map eventType %s to %s or %s in ContractNegotiationEventMessage", message.getType(),
-                        DSPACE_NEGOTIATION_PROPERTY_EVENT_TYPE_ACCEPTED, DSPACE_NEGOTIATION_PROPERTY_EVENT_TYPE_FINALIZED));
-                return null;
+                // this cannot happen
+                throw new AssertionError("Unknown event type: " + message.getType());
```

Comment:
Better to throw an `EdcException` as we have them everywhere else throughout the code base, even in "can-not-happen" situations